ext4: make ext4_xattr_list() static

From: Mingming Cao <cmm@us.ibm.com>

This patch makes the needlessly global ext4_xattr_list() static.

Signed-off-by: Mingming Cao <cmm@us.ibm.com>

---

 fs/ext4/xattr.c |    4 +++-
 fs/ext4/xattr.h |    7 -------
 2 files changed, 3 insertions(+), 8 deletions(-)

Index: linux-2.6.25-rc3/fs/ext4/xattr.c
===================================================================
--- linux-2.6.25-rc3.orig/fs/ext4/xattr.c	2008-03-03 15:52:35.000000000 -0800
+++ linux-2.6.25-rc3/fs/ext4/xattr.c	2008-03-03 15:52:44.000000000 -0800
@@ -92,6 +92,8 @@ static struct buffer_head *ext4_xattr_ca
 						 struct mb_cache_entry **);
 static void ext4_xattr_rehash(struct ext4_xattr_header *,
 			      struct ext4_xattr_entry *);
+static int ext4_xattr_list(struct inode *inode, char *buffer,
+			   size_t buffer_size);
 
 static struct mb_cache *ext4_xattr_cache;
 
@@ -420,7 +422,7 @@ cleanup:
  * Returns a negative error number on failure, or the number of bytes
  * used / required on success.
  */
-int
+static int
 ext4_xattr_list(struct inode *inode, char *buffer, size_t buffer_size)
 {
 	int i_error, b_error;
Index: linux-2.6.25-rc3/fs/ext4/xattr.h
===================================================================
--- linux-2.6.25-rc3.orig/fs/ext4/xattr.h	2008-03-03 11:45:59.000000000 -0800
+++ linux-2.6.25-rc3/fs/ext4/xattr.h	2008-03-03 15:52:44.000000000 -0800
@@ -74,7 +74,6 @@ extern struct xattr_handler ext4_xattr_s
 extern ssize_t ext4_listxattr(struct dentry *, char *, size_t);
 
 extern int ext4_xattr_get(struct inode *, int, const char *, void *, size_t);
-extern int ext4_xattr_list(struct inode *, char *, size_t);
 extern int ext4_xattr_set(struct inode *, int, const char *, const void *, size_t, int);
 extern int ext4_xattr_set_handle(handle_t *, struct inode *, int, const char *, const void *, size_t, int);
 
@@ -99,12 +98,6 @@ ext4_xattr_get(struct inode *inode, int 
 }
 
 static inline int
-ext4_xattr_list(struct inode *inode, void *buffer, size_t size)
-{
-	return -EOPNOTSUPP;
-}
-
-static inline int
 ext4_xattr_set(struct inode *inode, int name_index, const char *name,
 	       const void *value, size_t size, int flags)
 {
